Tests/SwiftGenTests/ConfigRunTests.swift

```diff
@@ -10,33 +10,88 @@ import Yams
 
 class ConfigRunTests: XCTestCase {
   private func _testRun(
-    fixture: String
-  ) throws {
+    fixture: String,
+    expected expectedLogs: [(LogLevel, String)],
+    unwanted unwantedLevels: Set<LogLevel> = [],
+    assertion assertionMessage: String,
+    file: StaticString = #file,
+    line: UInt = #line
+  ) {
     guard let path = Bundle(for: type(of: self)).path(forResource: fixture, ofType: "yml") else {
       fatalError("Fixture \(fixture) not found")
     }
-
+    var missingLogs = expectedLogs
     let configFile = Path(path)
-    let config = try Config(file: configFile)
-
-    try configFile.parent().chdir {
-      try config.runActions(verbose: true)
+    let logQueue = DispatchQueue(label: "swiftgen.log.queue")
+    let logger = { (level: LogLevel, msg: String) -> Void in
+      logQueue.async {
+        if let idx = missingLogs.firstIndex(where: { $0 == level && $1 == msg }) {
+          missingLogs.remove(at: idx)
+        } else if unwantedLevels.contains(level) {
+          XCTFail("Unexpected log: \(msg)")
+        }
+      }
+    }
+    let errorHandler = { (error: Error) -> Void in
+      if let idx = missingLogs.firstIndex(where: { $0 == .error && $1 == String(describing: error) }) {
+        missingLogs.remove(at: idx)
+      } else {
+        XCTFail("Unexpected error: \(error)", file: file, line: line)
+      }
     }
```

Comment:
nit: wondering if we could extract the `logger` and `errorHandler` closures into private methods to make that `_testRun` method shorter and more readable? Though they capture `missingLogs` (r/w) and `unwantedLevels` (readonly) variables so might not be as easy – except if we pass the `missingLogs` as `inout` 🤔 

Even if not extracting them out of the `_testRun` function, making them `func` instead of closures might make it more readable when human-parsing the code structure? 🤷 


Also, it could be a good occasion to rename that `_testRun` private func to something like `runTest()`?

Reply:
I'll look into it, just based myself on the existing config lint test code 🤷